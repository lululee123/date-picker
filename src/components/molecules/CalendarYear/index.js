import React, { useState, useEffect } from 'react';
import dayjs from 'dayjs';

import Day from 'components/atoms/Day';

import styles from './index.scss';

const CalendarYear = ({ data = [], selected, onClick }) => {
	const [years, setYears] = useState([]);

	useEffect(() => {
		const list = [];

		for (let i = parseInt(data[0], 10); i <= parseInt(data[1], 10); i += 1) {
			list.push(i.toString());
		}

		setYears(list);
	}, [data]);

	return (
		<div className={styles.calendarDay}>
			<Day
				value={dayjs(data[0]).subtract(1, 'year').format('YYYY')}
				displayText={dayjs(data[0]).subtract(1, 'year').format('YYYY')}
				className={styles.year}
				type="inActive"
				onClick={() => onClick(dayjs(data[0]).subtract(1, 'year').format('YYYY'))}
			/>
			{years.map(year => (
				<Day
					key={year}
					value={year}
					displayText={year}
					className={styles.year}
					type={year === selected ? 'active' : 'normal'}
					onClick={() => onClick(year)}
				/>
			))}
			<Day
				value={dayjs(data[1]).add(1, 'year').format('YYYY')}
				displayText={dayjs(data[1]).add(1, 'year').format('YYYY')}
				className={styles.year}
				type="inActive"
				onClick={() => onClick(dayjs(data[1]).add(1, 'year').format('YYYY'))}
			/>
		</div>
	);
};

export default CalendarYear;
