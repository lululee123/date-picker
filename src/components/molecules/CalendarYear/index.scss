.year {
    width: 39px;
    height: 39px;
    margin: 10px;
}