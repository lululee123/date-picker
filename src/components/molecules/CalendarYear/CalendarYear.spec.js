import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import store from 'store';

import CalendarYear from './index';

it('render CalendarYear correctly', () => {
	const component = renderer
		.create(
			<Provider store={store}>
				<CalendarYear />
			</Provider>,
		)
		.toJSON();
	expect(component).toMatchSnapshot();
});
