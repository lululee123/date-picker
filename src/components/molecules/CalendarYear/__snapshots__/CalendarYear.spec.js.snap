// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render CalendarYear correctly 1`] = `
<div
  className="calendarDay"
>
  <button
    className="day year inActive"
    onClick={[Function]}
    type="button"
  >
    2020
  </button>
  <button
    className="day year inActive"
    onClick={[Function]}
    type="button"
  >
    2022
  </button>
</div>
`;
